### PR TITLE
Serializer attributes options

### PR DESCRIPTION
**Pull request by nicklandgrebe**
#### Purpose

Add options to `Serializer.attributes` so we can apply conditions to multiple attributes at once.

#### Changes

* Extract `options` hash from last arg of `Serializer.attributes` using `ActiveSupport::Array#extract_options!`
  * Exclude `options[:key]`
* Pass in `options` to the subsequent `each` call to `attribute` => `attribute(attr, options)` 

#### Related GitHub issues

Old PR: #1714 

#### Additional helpful information

Resubmitted against 0-10-stable

### CODE REVIEW

**Comment by charlesvallieres** [comment]
Hi @nicklandgrebe I opened #2145 a couple days ago about the same issue. I think the extract options is good. If you could take the documentation update here : https://github.com/rails-api/active_model_serializers/pull/2145/files#diff-683b15b4d7c82f436aaeada1733d9eec and I'll close my PR so this can get merged upstream ASAP! :)

**Comment by nicklandgrebe** [comment]
@charlesvallieres Awesome, thanks so much for adding that useful doc

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -199,10 +199,12 @@ def self._attributes
     #   class AdminAuthorSerializer < ActiveModel::Serializer
     #     attributes :id, :name, :recent_edits
     def self.attributes(*attrs)
```
Comment (beauby) [inline lib/active_model/serializer.rb]: Why not `def self.attributes(*attrs, **options)`?

Comment (nicklandgrebe) [inline lib/active_model/serializer.rb]: That works! The things you learn. Thanks

**Comment by nicklandgrebe** [comment]
@bf4 The original use case of this feature was to be able to use one serializer in two different contexts of our API, separated between being queries made in a private environment (our own backend) vs. a public environment (a third party user). For example, a `Product`:

```
class ProductSerializer < ActiveModel::Serializer
  attributes :price, :number_available

  attributes :number_sold, :other_private_information, unless: :public?

  def public?
    # does the scope say we are in a public API environment?
  end
end

**Comment by maxKimoby** [comment]
Please post an update on this PR, not being able to use conditions on attributes makes life hard right now.

**Comment by nerfologist** [comment]
Hello, this would still be useful to us. Any reason why it has been closed?

Thanks

**Comment by tfwright** [comment]
Bump 🙏

**Comment by hashwin** [comment]
@bf4 Can this be followed up on? Looks like the changes were approved, but for some reason this was closed.

**Comment by btrd** [comment]
@bf4 Hi, I saw that you just released a new version, can you add this also in the next version ?

Thanks for your work !

**Comment by nakwa** [comment]
@bf4 @nicklandgrebe @beauby was there ever any update on this? 
I have this "temporarily" patched in my code for the last 4 years ... 😆

**Comment by sergioisidoro** [comment]
This was approved, but then closed? What happened?

**Pull request closed without merging**